elements/table-title-field.js

```diff
@@ -0,0 +1,163 @@
+var microcomponent = require('microcomponent')
+var nanomorph = require('nanomorph')
+var html = require('choo/html')
+
+module.exports = TitleField
+
+// Creates an input field with an explicit save button.
+// There's 2 modes: active and inactive.
+// Only dats that you're the owner of can have an active input field.
+// Inactive becomes active by clicking on the input field.
+// Active becomes inactive by:
+// - clicking anywhere outside the field
+// - pressing escape
+// - pressing enter (saving)
+// - clicking the save button (saving)
+function TitleField () {
+  var emit = null
+  var state = {
+    isEditing: false,
+    editTarget: null,
+    editValue: '',
+    title: null,
+    key: null
+  }
+
+  var component = microcomponent('table-row')
+  component.on('render', render)
+  component.on('render:active', renderActive)
+  component.on('render:inactive', renderInactive)
+  component.on('update', update)
+  component.on('unload', unload)
+  return component
+
+  function unload () {
+    emit = null
+    state = {
+      isEditing: false,
+      editTarget: null,
+      editValue: null,
+      isOwner: false,
+      title: null,
+      key: null
+    }
+  }
+
+  function update () {
+    var res = true
+    return res
+  }
+
+  function render (newDat, newState, newEmit) {
+    if (newEmit) emit = newEmit
+    if (newDat) {
+      state.isOwner = newDat.isOwner
+      state.key = newDat.key.toString('hex')
+      state.title = newDat.metadata.title || '#' + state.key
+    }
+
+    if (state.isEditing) return component.emit('render:active')
+    else return component.emit('render:inactive')
+  }
+
+  function renderInactive () {
+    state.editValue = ''
+
+    return html`
+      <section>
+        <h2 class="f6 normal truncate" onclick=${onclick}>
+          ${state.title}
+        </h2>
+      </section>
+    `
+
+    function onclick (e) {
+      e.stopPropagation()
+      e.preventDefault()
+      state.isEditing = true
+      component.emit('render')
+    }
+  }
+
+  function renderActive () {
+    if (!state.isEditing) {
+      state.isEditing = true
+      attachListener()
+    }
+
+    setTimeout(function () {
+      self._element.querySelector('input').focus()
+    }, 0)
```

Comment:
`on-load` wouldn't work here b/c we can't nest it at the moment. With you tho, I wish it worked as I'm not a fan of this pattern either. Looked into it last night and think we just need to debug `on-load`